ParabolInc/parabol
# feat(prompt-to-join): Add "would you like to join org" notification

Fixes #7792

<img width="750" alt="image" src="https://user-images.githubusercontent.com/466991/230328450-92726d32-1fb7-4378-801f-3fc359c5d3ec.png">

<img width="453" alt="image" src="https://user-images.githubusercontent.com/466991/230328647-2355822c-be46-4b59-a0d8-3b9892047bba.png">

<img width="480" alt="image" src="https://user-images.githubusercontent.com/466991/230328867-ab4f0b46-993e-4bf4-9bf1-9deb2df566a4.png">

**How to test:**

- Choose a custom domain your never used before. Or just erase it from `Organization.activeDomain` everywhere:

```
r.db('actionDevelopment').table('Organization')
  .update({
    activeDomain: null
});
```

- You should be able to receive email on it
- Enable email verification for the domain (Add an entry to the `OrganizationApprovedDomain` postgres table)
- Create a new account using this domain (you should receive a verification email that allows you to create an account)
- See that **no** prompt to join org notification received
- Add any other account to that org
- Add a feature flag to the org:

(Adds to all orgs)
```
r.db('actionDevelopment').table('Organization')
  .update({
    featureFlags: r.row('featureFlags').default([]).setInsert('promptToJoinOrg'),
});
```

- Create another account using this domain
- See prompt to join org notification appeared 
- Set tier `enterprise` for prev orgs on the domain

```
r.db('actionDevelopment').table('Organization')
  .filter({ activeDomain : 'PUT YOUR DOMAIN HERE'})  
  .update({
    tier: 'enterprise'
});
```

- Create another account using this domain, see no prompt
- Set tier `starter` back
- Disable email verification (`OrganizationApprovedDomain` table)
- Create another account using this domain, see no prompt
- Enable verification again, create account using **invite link**, see no prompt
- Sign up with gmail on that domain, see that prompt appeared
